metricbeat/module/redis/metricset.go

```diff
@@ -39,21 +42,38 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		IdleTimeout time.Duration `config:"idle_timeout"`
 		Network     string        `config:"network"`
 		MaxConn     int           `config:"maxconn" validate:"min=1"`
-		Password    string        `config:"password"`
 	}{
-		Network:  "tcp",
-		MaxConn:  10,
-		Password: "",
+		Network: "tcp",
+		MaxConn: 10,
 	}
 	err := base.Module().UnpackConfig(&config)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to read configuration")
 	}
 
+	uriParsed, _ := url.Parse(base.HostData().URI)
+	if uriParsed.RawQuery != "" {
+		queries := strings.Split(uriParsed.RawQuery, "&")
+		pw := ""
+		db := 0
+		for _, query := range queries {
+			querySplit := strings.Split(query, "=")
+			if querySplit[0] == "password" {
+				pw = querySplit[1]
+			} else if querySplit[0] == "db" {
+				db, _ = strconv.Atoi(querySplit[1])
```

Comment:
Should we fail if the `db` is not actually a number? If not, then we are defaulting to 0.

Reply:
we should not, if `db` is passed in as a string instead of int, we should return error here. I will this in next commit. Thanks!